### PR TITLE
cleanup: more compact gcode/mcode

### PR DESCRIPTION
**Pull request by gudnimg**
Remove 70 lines of code by making
the gcode/mcode parsing more compact in a few selected places

We can probably do a lot more but this is a small step the right direction.

No change in memory

### CODE REVIEW

File: Firmware/Marlin_main.cpp [hunk 1]
```diff
@@ -5427,17 +5424,18 @@ void process_commands()
...
       if(strchr_pointer>namestartpos)
         call_procedure=false;  //false alert, 'P' found within filename
 
       if( card.cardOK )
       {
         card.openFileReadFilteredGcode(namestartpos,!call_procedure);
-        if(code_seen('S'))
-          if(strchr_pointer<namestartpos) //only if "S" is occuring _before_ the filename
-            card.setIndex(code_value_long());
+        if(code_seen('S') && strchr_pointer < namestartpos) {
```
Comment (3d-gussner) [inline Firmware/Marlin_main.cpp:5435]: Why `{}` here while other places removed?

Comment (3d-gussner) [reply Firmware/Marlin_main.cpp:5435]: The comment can be above

File: Firmware/Marlin_main.cpp [hunk 2]
```diff
@@ -7306,9 +7254,10 @@ SERIAL_PROTOCOLPGM("\n\n");
       float temp = 150.0;
       int e = 0;
       int c = 5;
-      if (code_seen('E')) e = code_value_short();
-        if (e < 0)
-          temp = 70;
+      if (code_seen('E')) {
```
Comment (3d-gussner) [inline Firmware/Marlin_main.cpp:7257]: Why not?
if (code_seen('E')) e = code_value_short();
if (e < 0) temp = 70;

**Review by 3d-gussner — COMMENTED** [review]
See comments

**Review by DRracer — CHANGES_REQUESTED** [review]
While doing such a refactoring, please reformat the code segments according to our code style rules. Especially the "parsing" code is horrible and often mis-indented.

**Comment by gudnimg** [comment]
Closing this PR, we're probably better of just reformatting the entire file

**Pull request closed without merging**